SSA/Projects/RISCV64/Syntax.lean

```diff
@@ -0,0 +1,676 @@
+  Expr.mk
+    (op := Op.xor)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def sll  {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.sll)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def sra  {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.sll)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def mul {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.mul)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def div {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.div)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def divu {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.divu)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def remu {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.remu)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+def rem {Γ : Ctxt _} (e₁ e₂: Ctxt.Var Γ .bv) : Expr RV64 Γ .pure .bv  :=
+  Expr.mk
+    (op := Op.rem)
+    (eff_le := by constructor)
+    (ty_eq := by rfl)
+    (args := .cons e₁ <| .cons e₂ .nil)
+    (regArgs := HVector.nil)
+
+end RISCVExpr
+
+namespace RiscvMkExpr
+-- string representation of MLIR type into corresponding RISCV type
+def mkTy : MLIR.AST.MLIRType φ → MLIR.AST.ExceptM RV64 RV64.Ty
+  | MLIR.AST.MLIRType.undefined s => do
+    match s with
+    | "i64" => return .bv
+    | _ => throw .unsupportedType
+  | _ => throw .unsupportedType
+
+instance instTransformTy : MLIR.AST.TransformTy RV64 0 where
+  mkTy := mkTy
+
+def mkExpr (Γ : Ctxt _) (opStx : MLIR.AST.Op 0) :
+  MLIR.AST.ReaderM (RV64) (Σ eff ty, Expr (RV64) Γ eff ty) := do
+    match opStx.args with
+    | []  => do
+        match opStx.name with
+        | "li" => do
+            let some att := opStx.attrs.getAttr "imm"
+              | throw <| .unsupportedOp s!"no attribute in li, need to specify immediate {repr opStx}"
+            match att with
+              | .int val ty =>
+                let opTy@(.bv) ← mkTy ty
+                return ⟨.pure, opTy, ⟨
+                  .li (val),
+                  by
+                  simp[DialectSignature.outTy, signature]
+                ,
+                  by constructor,
+                  .nil,
+                  .nil
+                ⟩⟩
```

Comment:
Your formatting is off here, please check the rest of the code also.
Note: convention is to keep a space between `simp` and its lemmas `[...]`
```suggestion
                return ⟨.pure, opTy, ⟨
                  .li (val),
                  by simp [DialectSignature.outTy, signature],
                  by constructor,
                  .nil,
                  .nil
                ⟩⟩
```